### PR TITLE
Client

### PR DESCRIPTION
**Pull request by AntoninLaudon**

### CODE REVIEW

File: Client/CMakeLists.txt [hunk 1]
```diff
@@ -3,23 +3,24 @@ project(c_SnakeS)
 
 set(CMAKE_CXX_STANDARD 11)
 
-add_compile_options(-Wall -Wextra -Wpedantic -Werror -fno-gnu-unique)
+add_compile_options(-Wall -Wextra -Wpedantic -Werror -fno-gnu-unique -g)
```
Comment (Vivelis) [inline Client/CMakeLists.txt:6]: debug flag ?

Comment (AntoninLaudon) [reply Client/CMakeLists.txt:6]: oups valgrind

File: Client/src/Core.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#include "Core.hpp"
+
+Core::Core(std::pair<uint, uint> windowSize, std::pair<uint, uint> gameSize, std::pair<std::string, ushort> ipPort) {
+    _windowSize = windowSize;
+    _gameSize = gameSize;
+    _ipPort = std::make_shared<std::pair<std::string, ushort>>(ipPort);
+}
+
+Core::Core() {}
```
Comment (Vivelis) [inline Client/src/Core.cpp:9]: Did we need no parameter constructor ?

File: Client/src/Core.hpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#include <SFML/Graphics.hpp>
+#include <iostream>
+
+#include "Game.hpp"
+#include "Lobby.hpp"
+
+class Core {
+    public:
+        Core(std::pair<uint, uint> windowSize, std::pair<uint, uint> gameSize, std::pair<std::string, ushort> ipPort);
+        Core();
```
Comment (Vivelis) [inline Client/src/Core.hpp:13]: Did we need no parameter constructor ?

File: Client/src/Core.hpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        ~Core();
+
+        void setWindowSize(std::pair<uint, uint> windowSize);
+        void setGameSize(std::pair<uint, uint> gameSize);
+        void setIpPort(std::pair<std::string, ushort> ipPort);
+
+
```
Comment (Vivelis) [inline Client/src/Core.hpp:20]:
```suggestion

```

File: Client/src/Core.cpp (hunk 2 above)
Comment (AntoninLaudon) [reply Client/src/Core.cpp:9]: alors c'etait pour le definir PUIS le setup dans le main, sinon je peux pas le definir a la premiere ligne

File: Client/src/Game.cpp [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    _ipPort = ipPort;
+}
+
+std::pair<std::shared_ptr<sf::Texture>, sf::Sprite> Game::_loadSprite(const std::string &path) {
+    std::shared_ptr<sf::Texture> texture = std::make_shared<sf::Texture>();
+    sf::Sprite sprite;
+    
+    if (!texture->loadFromFile(path)) {
+        std::cerr << "Failed to load texture from " << path << std::endl;
+    }
```
Comment (Vivelis) [inline Client/src/Game.cpp:25]: shoold set texture to default value or return

File: Client/src/Lobby.cpp [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+        // draw a cancel button
+        sf::RectangleShape cancelButton(sf::Vector2f(50 * _scale, 12.5 * _scale));
+        sf::Text cancelButtonText("Cancel", _font, 8 * _scale);
+
+        cancelButtonText.setFillColor(sf::Color::White);
+        cancelButtonText.setPosition(36 * _scale, 80 * _scale);
+        cancelButton.setPosition(25 * _scale, 79 * _scale);
+        cancelButton.setOutlineColor(sf::Color::White);
+        cancelButton.setOutlineThickness(10);
```
Comment (Vivelis) [inline Client/src/Lobby.cpp:121]: I you can comment in code, you can create function.
Rule 42 of the programmer.

File: Client/src/Lobby.cpp [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+                default:
+                    break;
+            }
+        }
+        if (_event.type == sf::Event::MouseButtonPressed) {
+            if (_event.mouseButton.button == sf::Mouse::Left) {
+                _lastClick = _mousePos;
+                _is_clicking = true;
+            }
+        }
```
Comment (Vivelis) [inline Client/src/Lobby.cpp:172]: could use switch case instead of if
switch (_event.type)

File: Client/src/Lobby.hpp [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        sf::Vector2i _mousePos;
+        sf::Vector2i _lastClick;
+        bool _is_clicking;
+
+        // main window
+
+        // connecting window
+        bool _failed_connection;
+
+
```
Comment (Vivelis) [inline Client/src/Lobby.hpp:58]: void

File: Client/src/Tools.cpp [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    // Check if the user provided the correct number of arguments
+    if (argc != 3) {
+        std::cerr << "Usage: " << argv[0] << " -h" << std::endl;
+        return std::make_pair("__EXIT__", 1);
+    }
+
+    // Check if the IP address is valid
+    std::regex ipRegex("^((25[0-5]|(2[0-4]|1\\d|[1-9]|)\\d)(\\.(?!$)|$)){4}$");
```
Comment (Vivelis) [inline Client/src/Tools.cpp:25]: just here to say regex are good

File: Client/src/Tools.cpp [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        return std::make_pair("__EXIT__", 1);
+    }
+
+    // Check if the port number is valid
+    try {
+        int port = std::stoi(argv[2]);
+        if (port < 0 || port > 65535) throw std::invalid_argument("Invalid port");
+    } catch (std::invalid_argument &e) {
+        std::cerr << "Invalid port, expected a number between 0 and 65535" << std::endl;
+        return std::make_pair("__EXIT__", 1);
```
Comment (Vivelis) [inline Client/src/Tools.cpp:38]: hmmmmm weird but ok

**Review by Vivelis — CHANGES_REQUESTED** [review]

**Review by Vivelis — APPROVED** [review]

**Pull request merged**